Sources/REventLogger/Networking/ConfigureUrlRequest.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+
+protocol ConfigureUrlRequest {
+    var method: HTTPMethod { get }
+    var headers: [String: String]? { get }
+    var body: Encodable? { get }
+    var path: String { get }
+    func createURLRequest(with url: URL, body: [REvent]?) -> URLRequest?
+}
+
+extension ConfigureUrlRequest {
+    func createURLRequest(with url: URL, body: [REvent]?) -> URLRequest? {
+        guard let url = buildUrlFrom(baseUrl: url) else {
+            return nil
+        }
+        var request = URLRequest(url: url)
+        request.httpMethod = method.rawValue
+        if let body = body {
+            do {
+                let encoder = JSONEncoder()
+                encoder.keyEncodingStrategy = .convertToSnakeCase
+                request.httpBody = try encoder.encode(body)
+            } catch {
+                return nil
+            }
+        }
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        headers?.forEach { header in
+            request.setValue(header.value, forHTTPHeaderField: header.key)
+        }
+        return request
+    }
+
+    var headers: [String: String]? {
```

Comment:
is this default implementation needed?

Reply:
its needed beccause we will not always require a header to make a request

Reply:
I feel it is better to add when this is necessary. Like when we have a request that doesn't require header.